prism/src/userinterface/GUIPrism.java

```diff
@@ -405,12 +406,35 @@ public void windowClosing(java.awt.event.WindowEvent evt)
 	public void passCLArgs(String args[])
 	{
 		// just before we get started, pass any command-line args to all plugins
-		// we first remove the -javamaxmem/-javastack arguments, if present
+		// we first remove any -javamaxmem/-javastack arguments (ignored)
+		// and any -dir argument (treated globally)
 		List<String> argsCopy = new ArrayList<String>();
 		for (int i = 0; i < args.length; i++) {
-			if (args[i].equals("-javamaxmem") || args[i].equals("-javastack")) {
+			String sw = args[i].substring(1);
```

Comment:
Here, a check that the initial character is indeed a '-' is missing. E.g., running `xprism adir ..` will be interpreted as if it were `-dir ..`